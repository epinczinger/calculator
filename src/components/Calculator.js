import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import Calculate from '../logic/Calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = buttonName => {
    setState(state => Calculate(state, buttonName));
  };

  const { total, next, operation } = state;

  return (
    <>
      <div className="flex flex-col h-full px-3">
        <h1 className="text-gray-800 font-bold text-xl py-3">Let`s do some Maths</h1>
        <div className="mx-auto bg-gray-800">
          <Display total={total} next={next} operation={operation} />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </>
  );
};

export default Calculator;
