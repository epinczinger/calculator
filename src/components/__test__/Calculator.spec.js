import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Calculator component', () => {
  it('renders as expected matching snapshot', () => {
    const CalculatorTest = renderer.create(<Calculator />).toJSON();
    expect(CalculatorTest).toMatchSnapshot();
  });
});
