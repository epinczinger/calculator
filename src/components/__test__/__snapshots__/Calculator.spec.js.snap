// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders as expected matching snapshot 1`] = `
<div
  className="flex flex-col px-3"
>
  <h1
    className="text-gray-800 font-bold text-xl py-3 animate-pulse"
  >
    Let\`s do some Maths
  </h1>
  <div
    className="mx-auto bg-gray-800"
  >
    <div
      className="flex justify-end tracking-tighter bg-gray-400 border-4 w-64 border-gray-800 rounded-md text-lg lg:w-auto lg:text-2xl px-2 py-3 "
    >
      <h2
        className="text-gray-700"
      >
        0
      </h2>
    </div>
    <div
      className="bg-gray-800 text-lg md:text-xl"
    >
      <div
        className="flex"
      >
        <div
          className="bg-blue-700 hover:bg-blue-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
        </div>
        <div
          className="bg-blue-700 hover:bg-blue-800 w-1/4 flex justify-center border rounded border-gray-800 min-w-12"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
        </div>
        <div
          className="bg-blue-700 hover:bg-blue-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </div>
        <div
          className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
      </div>
      <div
        className="flex"
      >
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </div>
        <div
          className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
      </div>
      <div
        className="flex"
      >
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
        </div>
        <div
          className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
      </div>
      <div
        className="flex"
      >
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </div>
        <div
          className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <div
        className="flex"
      >
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-32 h-16 lg:w-48 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </div>
        <div
          className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
        <div
          className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
        >
          <button
            className="w-full h-full"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
